cmd/update-main.go

```diff
@@ -68,90 +68,108 @@ const (
 	minioReleaseURL           = "https://dl.minio.io/server/minio/release/" + runtime.GOOS + "-" + runtime.GOARCH + "/"
 )
 
-func getCurrentReleaseTime(minioVersion, minioBinaryPath string) (releaseTime time.Time, err error) {
-	if releaseTime, err = time.Parse(time.RFC3339, minioVersion); err == nil {
+// updateCheckParams - collection of parameters used to check if the
+// version of the current server process is older than the latest
+// available official release. They enable the testability of various
+// routines in the update checking process.
+type updateCheckParams struct {
+	// version of the current server process
+	version string
+	// path to the server binary
+	serverBinaryPath string
+	// file on the server to check for docker env
+	cgroupFile string
+	// URL of release checksum file (contains version of latest
+	// release)
+	releaseChecksumURL string
+	// current process run mode
+	serverMode string
+}
+
+// getCurrentReleaseTime - returns this process's release time.  If it
+// is official minio version, parsed version is returned else minio
+// binary's mod time is returned.
+func (u *updateCheckParams) getCurrentReleaseTime() (releaseTime time.Time,
```

Comment:
* As you told WRT `updateCheckParams` is not performance sensitive. the pointer type is good for methods changing fields of the type.  This case is applicable to the struct as none of the methods are updating fields.

* As I indicated in the example, pointer type must be initialized to avoid run time panic.  This is avoidable if methods are to value type.

* method to pointer type is even accessible by `nil` pointer.   Having any methods to pointer type will lead to panic.
```go
func (t *mytype) greet() {
	fmt.Println("spock hand")
}

var t2 *mytype
t1.greet()
```

* Choosing pointer or value type for method is by considering performance and/or changing field values.